sklearn/decomposition/_nmf.py

```diff
@@ -1437,3 +1611,358 @@ def inverse_transform(self, W):
         """
         check_is_fitted(self)
         return np.dot(W, self.components_)
+
+
+class MiniBatchNMF(NMF):
+    """Mini-Batch and online Non-Negative Matrix Factorization (NMF)
+
+    .. versionadded:: 1.0
+
+    Find two non-negative matrices (W, H) whose product approximates the non-
+    negative matrix X. This factorization can be used for example for
+    dimensionality reduction, source separation or topic extraction.
+
+    The objective function is:
+
+        .. math::
+
+            0.5 * ||X - WH||_{loss}^2 + alpha * l1_{ratio} * ||vec(W)||_1
+
+            + alpha * l1_{ratio} * ||vec(H)||_1
+
+            + 0.5 * alpha * (1 - l1_{ratio}) * ||W||_{Fro}^2
+
+            + 0.5 * alpha * (1 - l1_{ratio}) * ||H||_{Fro}^2
+
+    Where:
+
+    :math:`||A||_{Fro}^2 = \\sum_{i,j} A_{ij}^2` (Frobenius norm)
+
+    :math:`||vec(A)||_1 = \\sum_{i,j} abs(A_{ij})` (Elementwise L1 norm)
+
+    The generic norm :math:`||X - WH||_{loss}^2` may represent
+    the Frobenius norm or another supported beta-divergence loss.
+    The choice between options is controlled by the `beta_loss` parameter.
+
+    The objective function is minimized with an alternating minimization of W
+    and H.
+
+    Read more in the :ref:`User Guide <MiniBatchNMF>`.
+
+    Parameters
+    ----------
+    n_components : int or None
+        Number of components, if n_components is not set all features
+        are kept.
+
+    init : {'random', 'nndsvd', 'nndsvda', 'nndsvdar', 'custom'}, default=None
+        Method used to initialize the procedure.
+        Default: None.
+        Valid options:
+
+        - `None`: 'nndsvd' if n_components <= min(n_samples, n_features),
+          otherwise random.
+
+        - `'random'`: non-negative random matrices, scaled with:
+          sqrt(X.mean() / n_components)
+
+        - `'nndsvd'`: Nonnegative Double Singular Value Decomposition (NNDSVD)
+          initialization (better for sparseness)
+
+        - `'nndsvda'`: NNDSVD with zeros filled with the average of X
+          (better when sparsity is not desired)
+
+        - `'nndsvdar'` NNDSVD with zeros filled with small random values
+          (generally faster, less accurate alternative to NNDSVDa
+          for when sparsity is not desired)
+
+        - `'custom'`: use custom matrices W and H
+
+    batch_size : int, default=1024
+        Number of samples in each mini-batch. Large batch sizes
+        give better long-term convergence at the cost of a slower start.
+
+    solver : 'mu'
+        Numerical solver to use:
+        'mu' is a Multiplicative Update solver.
+        For now, this is the only available solver in the
+        MiniBatch implementation.
```

Comment:
Is that so unrealistic to imagine of implementing a new solver one day?

Reply:
maybe one day but I don't see this coming in the near future and I don't think we want to introduce parameters for a possible enhancement we are not even sure it will exist at some point. Besides, if we introduce a new solver one day, we'll add a new parameter then.

Reply:
Should this parameter be removed, then?